apps/api/src/app/integrations/e2e/create-integration.e2e.ts

```diff
@@ -48,57 +49,120 @@ describe('Create Integration - /integration (POST)', function () {
+      expect(integration.active).to.equal(payload.active);
+    }
   });
 
-  it('should fail due missing param', async function () {
+  it('should not allow to create integration with same identifier', async function () {
     const payload = {
-      providerId: 'sendgrid',
-      channel: 'email',
-      active: true,
+      providerId: EmailProviderIdEnum.SendGrid,
+      channel: ChannelTypeEnum.EMAIL,
+      identifier: 'identifier',
+      active: false,
       check: false,
     };
+    await integrationRepository.create({
+      name: 'Test',
+      identifier: payload.identifier,
+      providerId: EmailProviderIdEnum.SendGrid,
+      channel: ChannelTypeEnum.EMAIL,
+      active: false,
+      _organizationId: session.organization._id,
+      _environmentId: session.environment._id,
+    });
 
-    const body = (await session.testAgent.post('/v1/integrations').send(payload)).body.data as IntegrationEntity;
+    const { body } = await session.testAgent.post('/v1/integrations').send(payload);
 
-    expect(body).to.equal(undefined);
+    expect(body.statusCode).to.equal(400);
+    expect(body.message).to.equal('Integration with identifier already exists');
   });
 
-  it('should deactivated old providers', async function () {
+  it('should not allow to activate the integration without the credentials', async function () {
     const payload = {
-      providerId: 'mailgun',
-      channel: 'email',
-      credentials: { apiKey: '123', secretKey: 'abc' },
+      providerId: EmailProviderIdEnum.SendGrid,
+      channel: ChannelTypeEnum.EMAIL,
       active: true,
       check: false,
     };
 
-    const environmentId = (await session.testAgent.get(`/v1/integrations`)).body.data[0]._environmentId;
+    const { body } = await session.testAgent.post('/v1/integrations').send(payload);
+
+    expect(body.statusCode).to.equal(400);
+    expect(body.message).to.equal('The credentials are required to activate the integration');
+  });
+
+  it('should allow creating the integration with minimal data', async function () {
+    const payload = {
+      providerId: EmailProviderIdEnum.SendGrid,
+      channel: ChannelTypeEnum.EMAIL,
+      check: false,
+    };
+
+    const {
+      body: { data },
+    } = await session.testAgent.post('/v1/integrations').send(payload);
 
-    await session.testAgent.post('/v1/integrations').send(payload);
+    expect(data.name).to.equal('SendGrid');
+    expect(data.identifier).to.exist;
+    expect(data.providerId).to.equal(EmailProviderIdEnum.SendGrid);
+    expect(data.channel).to.equal(ChannelTypeEnum.EMAIL);
+    expect(data.active).to.equal(false);
+  });
 
-    const integrations = await integrationRepository.findByEnvironmentId(environmentId);
+  it('should allow creating the integration for different environment', async function () {
+    const prodEnv = await envRepository.findOne({ name: 'Production', _organizationId: session.organization._id });
```

Comment:
I think this test misses to show that we already have the same entry but for development environment. We could even compare that the data in both are the same but the _id and the identifiers.

Reply:
Then I would rename the test to `should allow creating the integration in the chosen environment`. Current name makes me think an integration is created in one environment already (dev, considering the usage of `prodEnv`) and we are just checking we can create the same integration in a different one.